catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/FileSystemPersistenceProvider.java

```diff
@@ -215,7 +168,60 @@ public void clear() {
     }
   }
 
-  private File getMapStoreFile(String key) {
-    return new File(getMapStorePath() + key + SER);
+  Object loadFromPersistence(String key) {
+    File file = getMapStoreFilePath(key).toFile();
+    if (!file.exists()) {
+      return null;
+    }
+    InputStream inputStream = null;
+    try {
+      inputStream = new FileInputStream(Paths.get(persistencePath.toString(), key + SER).toFile());
+      InputStream buffer = new BufferedInputStream(inputStream);
+
+      try (ObjectInput input = new ObjectInputStream(buffer)) {
+        return input.readObject();
+      }
+    } catch (IOException e) {
+      LOGGER.info("Unable to read object.", e);
+    } catch (ClassNotFoundException e) {
+      LOGGER.info("Class for object being read from stream does not exist.", e);
```

Comment:
❓Wonder if we should consider deleting the file when this happens?

Reply:
Updated to delete in this case